ml-agents/mlagents/trainers/config.py

```diff
@@ -0,0 +1,58 @@
+import attr
+from typing import Dict
+
+
+@attr.s(auto_attribs=True)
+class RewardSignalParameters:
+    strength: float = 1.0
+    gamma: float = .99
+
+
+@attr.s(auto_attribs=True)
+class ExtrinsicRewardSignalParameters(RewardSignalParameters):
+    pass
+
+@attr.s(auto_attribs=True)
+class CuriosityRewardSignalParameters(RewardSignalParameters):
+    encoding_size: int = 256
+
+@attr.s(auto_attribs=True)
+class GailRewardSignalParameters(RewardSignalParameters):
+    pass
+
+
+reward_name_to_parameter_class = {
+    "extrinsic": ExtrinsicRewardSignalParameters,
+    "gail": GailRewardSignalParameters,
+    "curiosity": CuriosityRewardSignalParameters
+}
+
+@attr.s(auto_attribs=True)
+class TrainerParameters:
+    trainer: str = "ppo"  # TODO enum
+    batch_size: int = 1024
+    beta: float = 5.0e-3
+    buffer_size: int = 10240
+    epsilon: float = 0.2
+    gamma: float = .99  # TODO is this the current default?
```

Comment:
We can remove this as it's defined in the reward signals

Reply:
Ah, I think a few configs are still passing it in :)

Reply:
Uh oh, that's a bug - I removed them and pushed it to #2296